### PR TITLE
rbd: add snapshot count in rbd info

### PR DESCRIPTION
**Pull request by zy751713126**

### CODE REVIEW

**Comment by zy751713126** [comment]
retest this please

File: src/tools/rbd/action/Info.cc [hunk 1]
```diff
@@ -83,7 +83,12 @@ static int do_show_info(librados::IoCtx &io_ctx, librbd::Image& image,
   uint64_t overlap, features, flags, snap_limit;
   bool snap_protected = false;
   librbd::mirror_image_info_t mirror_image;
-  int r;
+  std::vector<librbd::snap_info_t> snaps;
+  int r, r1;
+
+  r1 = image.snap_list(snaps);
```
Comment (dillaman) [inline src/tools/rbd/action/Info.cc]: Nit: no need for a new ```r1``` variable

**Comment by dillaman** [comment]
@zy751713126 Can you rebase against the latest version of the master branch and repush?

**Comment by zy751713126** [comment]
I have rebased the latest version and repushed. 
thanks，dillaman.

**Comment by dillaman** [comment]
retest this please

**Comment by zy751713126** [comment]
retest this please

**Comment by dillaman** [comment]
@zy751713126 The formatted output test fails [1]:

[1] http://qa-proxy.ceph.com/teuthology/jdillaman-2018-04-13_12:10:29-rbd-wip-jd-testing-distro-basic-smithi/2394476/teuthology.log

File: src/test/cli-integration/rbd/formatted-output.t [hunk 2]
```diff
@@ -329,7 +347,9 @@ whenever it is run. grep -v to ignore it, but still work on other distros.
   rbd image 'quux':
   \tsize 1024 kB in 1 objects (esc)
   \torder 22 (4096 kB objects) (esc)
+  \tsnapshot_count: 0 (esc)
   [^^]+ (re)
+  \tblock_name_prefix: rb.0.* (glob)
```
Comment (dillaman) [inline src/test/cli-integration/rbd/formatted-output.t]: Nit: fails here

File: src/test/cli-integration/rbd/formatted-output.t [hunk 3]
```diff
@@ -350,15 +371,17 @@ whenever it is run. grep -v to ignore it, but still work on other distros.
...
     <format>1</format>
   </image>
   $ rbd info rbd_other/child
   rbd image 'child':
   \tsize 512 MB in 128 objects (esc)
   \torder 22 (4096 kB objects) (esc)
-  \tid:* (glob)
+  \tsnapshot_count: 1 (esc)
   [^^]+ (re)
+  \tblock_name_prefix: rb.0.* (glob)
```
Comment (dillaman) [inline src/test/cli-integration/rbd/formatted-output.t]: Nit: fails here

File: src/test/cli-integration/rbd/formatted-output.t [hunk 4]
```diff
@@ -408,8 +433,9 @@ whenever it is run. grep -v to ignore it, but still work on other distros.
   rbd image 'child':
   \tsize 512 MB in 128 objects (esc)
   \torder 22 (4096 kB objects) (esc)
-  \tid:* (glob)
+  \tsnapshot_count: 1 (esc)
   [^^]+ (re)
+  \tblock_name_prefix: rbd_data.* (glob)
```
Comment (dillaman) [inline src/test/cli-integration/rbd/formatted-output.t]: Nit: fails here

**Comment by zy751713126** [comment]
@dillaman hi, dillaman, all checks of this patch  have passed, could you help me merge it. thanks a lot.

**Comment by dillaman** [comment]
Still is failing the same test: http://qa-proxy.ceph.com/teuthology/jdillaman-2018-04-26_17:23:27-rbd-wip-jd-testing-distro-basic-smithi/2443289/teuthology.log

**Comment by zy751713126** [comment]
2018-05-10T00:32:07.207 INFO:tasks.cram.client.0.smithi145.stdout:@@ -59,7 +59,7 @@
2018-05-10T00:32:07.207 INFO:tasks.cram.client.0.smithi145.stdout:   \tsize 1GiB in 256 objects (esc)
2018-05-10T00:32:07.207 INFO:tasks.cram.client.0.smithi145.stdout:   \torder 22 (4MiB objects) (esc)
2018-05-10T00:32:07.207 INFO:tasks.cram.client.0.smithi145.stdout:   [^^]+ (re)
2018-05-10T00:32:07.207 INFO:tasks.cram.client.0.smithi145.stdout:-  \tblock_name_prefix: rb.0.*  (glob)
2018-05-10T00:32:07.207 INFO:tasks.cram.client.0.smithi145.stdout:+  \tblock_name_prefix: rb.0.1044.625558ec (esc)
2018-05-10T00:32:07.208 INFO:tasks.cram.client.0.smithi145.stdout:   \tformat: 1 (esc)
2018-05-10T00:32:07.208 INFO:tasks.cram.client.0.smithi145.stdout:   $ rbd info foo --format json | python -mjson.tool | sed 's/,$/, /'
2018-05-10T00:32:07.208 INFO:tasks.cram.client.0.smithi145.stdout:   {
2018-05-10T00:32:07.208 INFO:tasks.cram.client.0.smithi145.stdout:@@ -92,7 +92,7 @@
2018-05-10T00:32:07.208 INFO:tasks.cram.client.0.smithi145.stdout:   \tsize 1GiB in 256 objects (esc)
2018-05-10T00:32:07.208 INFO:tasks.cram.client.0.smithi145.stdout:   \torder 22 (4MiB objects) (esc)
2018-05-10T00:32:07.208 INFO:tasks.cram.client.0.smithi145.stdout:   [^^]+ (re)
2018-05-10T00:32:07.208 INFO:tasks.cram.client.0.smithi145.stdout:-  \tblock_name_prefix: rb.0.*  (glob)
2018-05-10T00:32:07.209 INFO:tasks.cram.client.0.smithi145.stdout:+  \tblock_name_prefix: rb.0.1044.625558ec (esc)
2018-05-10T00:32:07.209 INFO:tasks.cram.client.0.smithi145.stdout:   \tformat: 1 (esc)
2018-05-10T00:32:07.209 INFO:tasks.cram.client.0.smithi145.stdout:   \tprotected: False (esc)
2018-05-10T00:32:07.209 INFO:tasks.cram.client.0.smithi145.stdout:   $ rbd info foo@snap --format json | python -mjson.tool | sed 's/,$/, /'
2018-05-10T00:32:07.209 INFO:tasks.cram.client.0.smithi145.stdout:# Ran 1 tests, 0 skipped, 1 failed.

hi, @dillaman , I am a newcomer，I don‘t know how to write block_name_prefix output format，could you please help me to give directions, thanks a lot.

**Comment by dillaman** [comment]
Manually merged PR after fixing test.

**Comment by dillaman** [comment]
@zy751713126 The issue was that you added ```block_name_prefix``` lines to the test when you were should have added ```snapshot_count" lines.

**Comment by zy751713126** [comment]
thanks

**Pull request closed without merging**